charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -1687,13 +1687,21 @@ exporters:
 
 {{- define "opentelemetry-collector.applyCoralogixExporterConfig" -}}
 {{- $config := mustMergeOverwrite (include "opentelemetry-collector.coralogixExporterConfig" .Values | fromYaml) .config }}
-{{- $pipeline := "all" }}
-{{- with .Values.Values.presets.coralogixExporter.pipeline }}
-{{- $pipeline = . }}
+{{- $pipeline := list }}
+{{- with .Values.Values.presets.coralogixExporter.pipelines }}
+  {{- $pipeline = . }}
+{{- else }}
+  {{- with .Values.Values.presets.coralogixExporter.pipeline }}
+    {{- $pipeline = list . }}
+  {{- else }}
+    {{- $pipeline = list "all" }}
+  {{- end }}
```

Comment:
I think we can use `else with` here to simplify a little bit:

```suggestion
{{- $pipeline := list }}
{{- with .Values.Values.presets.coralogixExporter.pipelines }}
  {{- $pipeline = . }}
{{- else with .Values.Values.presets.coralogixExporter.pipeline }}
  {{- $pipeline = list . }}
{{- else }}
  {{- $pipeline = list "all" }}
{{- end }}
```

Got this idea while learning that `with` can have `else` (I didn't know about it):

```
{{with pipeline}} T1 {{else with pipeline}} T0 {{end}}
	To simplify the appearance of with-else chains, the else action
	of a with may include another with directly; the effect is exactly
	the same as writing
		{{with pipeline}} T1 {{else}}{{with pipeline}} T0 {{end}}{{end}}
```

From https://pkg.go.dev/text/template.

Reply:
@douglascamata sounds like a good idea, but when I try to apply this change, helm gives me a strange error

```
Error: Failed to render chart: exit status 1: Error: parse error at (opentelemetry-collector/templates/_config.tpl:1692): undefined variable "$pipeline"
```

<img width="975" alt="image" src="https://github.com/user-attachments/assets/f721c875-348e-4b76-a180-b34084b635bb" />

Reply:
@douglascamata nope, this won't work, since `with` has some weird behavior in changing the scope

but I've found a simpler solution that has the same logic

```
{{- $pipeline := list "all" }}
{{- if .Values.Values.presets.coralogixExporter.pipelines }}
  {{- $pipeline = .Values.Values.presets.coralogixExporter.pipelines }}
{{- else if .Values.Values.presets.coralogixExporter.pipeline }}
  {{- $pipeline = list .Values.Values.presets.coralogixExporter.pipeline }}
{{- end }}
```

hope that's fine :D